packages/cosmic-swingset/scripts/clean-core-eval.js

```diff
@@ -0,0 +1,71 @@
+#! /usr/bin/env node
+import '@endo/init';
+import * as farExports from '@endo/far';
+import { isEntrypoint } from '../src/is-entrypoint.js';
+
+export const compartmentEvaluate = code => {
+  // const permit = true;
+  // const powers = {};
+  const modules = {}; // TODO
+
+  // Inspired by ../repl.js:
+  const globals = harden({
+    ...modules,
+    ...farExports,
+    assert,
+    console: {
+      // Ensure we don't pollute stdout.
+      debug: console.warn,
+      log: console.warn,
+      info: console.warn,
+      warn: console.warn,
+      error: console.error,
+    },
+  });
+
+  // Evaluate the code in the context of the globals.
+  const compartment = new Compartment(globals);
+  harden(compartment.globalThis);
+  return compartment.evaluate(code);
+};
+
+export const htmlStartCommentPattern = new RegExp(`(${'<'})(!--)`, 'g');
+export const htmlEndCommentPattern = new RegExp(`(--)(${'>'})`, 'g');
+export const importPattern = new RegExp(
+  '(^|[^.])\\bimport(\\s*(?:\\(|/[/*]))',
+  'g',
+);
+
+// Neutralize HTML comments and import expressions.
+export const defangEvaluableCode = code =>
+  code
+    .replace(importPattern, '$1import\\$2') // avoid SES_IMPORT_REJECTED
+    .replace(htmlStartCommentPattern, '$1\\$2') // avoid SES_HTML_COMMENT_REJECTED
+    .replace(htmlEndCommentPattern, '$1\\$2'); // avoid SES_HTML_COMMENT_REJECTED
+
+export const main = async (argv, { readFile, stdout }) => {
+  const [_node, _script, fn] = argv;
+  if (fn === undefined) {
+    console.error(`Usage: ${_script} <filename>`);
+    process.exit(1);
+  }
+  const text = await readFile(fn, 'utf-8');
+  const clean = defangEvaluableCode(text);
+  const withURL = `${clean}\n//# sourceURL=${fn}\n`;
+
+  // end-of-line whitespace disrupts YAML formatting
+  const trimmed = withURL.replace(/[\r\t ]+$/gm, '');
+  compartmentEvaluate(trimmed);
+  stdout.write(trimmed);
+};
+
+if (isEntrypoint(import.meta.url)) {
```

Comment:
interesting approach. `isEntrypoint` is powerful, but I suppose so was `require.main`.

A [Stackoverflow post](https://stackoverflow.com/a/68848622/7963) has a succinct version:

```
if (import.meta.url === `file://${process.argv[1]}`) {
  // module was not imported but called directly
}
```

but that's less robust; it's followed by one that uses `url.pathToFileURL` as in `isEntrypoint`.